lib/redis/cluster.rb

```diff
@@ -69,7 +69,12 @@ def with_reconnect(val = true, &block)
     end
 
     def call(command, &block)
-      send_command(command, &block)
+      begin
+        send_command(command, &block)
+      rescue CannotConnectError => error
+        update_cluster_info!
+        raise error
```

Comment:
Please use a straight `raise` without argument, so that the original backtrace is preserved.

Reply:
Which also means you won't have to assign `error`